File: Lab 9/t1.go
package main

import (
  "fmt"
  "os"
  "bufio"
  "strings"
)

func main() {
  fmt.Println("Please input a line:")
  line, e := bufio.NewReader(os.Stdin).ReadString('\n')
  if e != nil {
    fmt.Println("Read error:", e)
    return
  }

  fmt.Println("There are", len(strings.Fields(line)), "words in the line.")
}

File: Lab 9/t1.rs
fn main() {
  println!("Please input a line:");

  let mut buf = String::new();
  match std::io::stdin().read_line(&mut buf) {
    Ok(_) => {}
    Err(e) => {
      eprintln!("Read error: {}", e);
      std::process::exit(1);
    }
  }

  println!("There are {} words in the line.",
           buf.split_whitespace().count());
}
